src/components/decorative-boxes/index.tsx

```diff
@@ -0,0 +1,155 @@
+import { useRef } from 'react';
+import style from './style.module.css';
+type BoxProperties = {
+  color: `--color-${DefaultColors}${ColorModifiers}` | HexColorAsString;
+  position: `${VerticalPosition}-${HorizontalPosition}` | NumericPosition;
+  text?: string | undefined;
+};
+
+type NumericPosition = {
+  horizontalPosition: number;
+  verticalPosition: number;
+};
+type BoxSize = 'small' | 'medium' | 'large' | number;
+type VerticalPosition = 'top' | 'topish' | 'middle' | 'bottomish' | 'bottom';
+type HorizontalPosition = 'left' | 'leftish' | 'middle' | 'rightish' | 'right';
+type DefaultColors = 'primary' | `secondary${1 | 2 | 3 | 4}`;
+type ColorModifiers = '' | `__${'tint' | 'shade'}${1 | 2 | 3 | 4}`;
+type HexColorAsString = `#${string}`;
+
+export interface DecorativeBoxesProps
+  extends React.ComponentPropsWithoutRef<any> {
+  children: JSX.Element;
+  boxSize?: BoxSize;
+  box1Properties: BoxProperties;
+  box2Properties: BoxProperties;
+}
+
+const DecorativeBoxes: React.FC<DecorativeBoxesProps> = ({
```

Comment:
Skal selvsagt ikke presse på mine preferanser, men etterlater denne her likevel jeg 😄  https://blog.variant.no/a-better-way-to-type-react-components-9a6460a1d4b7

Reply:
Hehe, bytta til function nå. Jeg har faktisk lest artikkelen din om det allerede, men tror det gikk i glemmeboka akkurat her.